### PR TITLE
Chore - TypeScript 5 upgrade 

### PR DESCRIPTION
**Pull request by devinea**
"typescript": "4.9.5" = > "5.3.3"
"@typescript-eslint/eslint-plugin": "5.62.0" => "6.21.0"
"@typescript-eslint/parser": "5.62.0" => "6.21.0"
"update-ts-references": "2.6.1" => "3.2.1"
"ts-jest": "29.1.1" => "29.1.2"
"ts-node": "10.9.1" => "10.9.2"

Adapt code for newer eslint and TS.

### CODE REVIEW

**Review by Klaus-Keller — APPROVED** [review]
Thanks @devinea!
- update of dependencies looks good
- linting looks good
- did no manual test, just visually

Approved from my side.

**Review by 815are — DISMISSED** [review]
Checked `fpm-writter`, `ui-components`:
1. did not find issue with lint fixes;
2. tested storybook and changed components - did not notice issue

File: packages/app-config-writer/src/prompt/smartlinks-config.ts [hunk 1]
```diff
@@ -146,7 +146,7 @@ export async function promptUserPass(log?: ToolsLogger): Promise<AxiosBasicCrede
                 name: 'password',
                 message: `${cyan(t('info.password'))}`,
                 validate: (value: string): boolean | string => {
-                    if (!value || !value.trim()) {
+                    if (!value?.trim()) {
```
Comment (ullasholla) [inline packages/app-config-writer/src/prompt/smartlinks-config.ts:149]: Are these modifications aimed at addressing linting errors? They aren't associated with the version upgrade. While they are necessary, I'm mindful of maintaining a distinction between the version upgrade and code clean-up.

Comment (devinea) [reply packages/app-config-writer/src/prompt/smartlinks-config.ts:149]: @ullasholla I know what you mean, I hadn't planned to update it originally. 
The @typescript-eslint/eslint-plugin version needed to be upgraded as the old one didn't support the new TypeScript version.

File: packages/ui5-config/src/defaults.ts [hunk 2]
```diff
@@ -2,11 +2,11 @@ export const enum UI5_DEFAULT {
     DEFAULT_UI5_VERSION = '',
     DEFAULT_LOCAL_UI5_VERSION = '1.95.0',
     MIN_UI5_VERSION = '1.60.0',
-    MIN_LOCAL_SAPUI5_VERSION = '1.76.0',
+    MIN_LOCAL_SAPUI5_VERSION = '1.76.0', // eslint-disable-line @typescript-eslint/no-duplicate-enum-values
     MIN_LOCAL_OPENUI5_VERSION = '1.52.5',
     SAPUI5_CDN = 'https://ui5.sap.com',
     OPENUI5_CDN = 'https://sdk.openui5.org',
-    TYPES_VERSION_SINCE = '1.76.0',
+    TYPES_VERSION_SINCE = '1.76.0', // eslint-disable-line @typescript-eslint/no-duplicate-enum-values
```
Comment (ullasholla) [inline packages/ui5-config/src/defaults.ts:9]: Why are the two eslint configuration comments needed?

**Review by ullasholla — CHANGES_REQUESTED** [review]
The version upgrades look good. Thanks @devinea 

Had a few questions:
1. The PR description needs to mention `ts-node` upgrade as well
2. There is a SonarCloud failure - https://github.com/SAP/open-ux-tools/pull/1641/checks?check_run_id=21275584191
3. The PR includes cleaning up of the code. Is that essential to the PR? The changes are good and necessary, but do they need to be part of this PR
4. The branch is out-of-date with the base branch (does updating invalidate the previous approvals?)

File: packages/ui5-config/src/defaults.ts (hunk 2 above)
Comment (devinea) [reply packages/ui5-config/src/defaults.ts]: Reduced to 1 in https://github.com/SAP/open-ux-tools/pull/1641/commits/fc71c65823caa490be5c62215e2882d96bcce10a

**Review by ullasholla — APPROVED** [review]
Thanks for addressing the review comments @devinea.

I think it's a bit unfair on you to try and increase test coverage as part of this PR - all the other issues are taken care of.

**Review by voicis — APPROVED** [review]
Checked CPE and annotation packages
- update of dependencies looks good
- did no manual test, just visually

**Comment by devinea** [comment]
Thanks all.

**Pull request merged**